### PR TITLE
Update System.Runtime.WindowsRuntime to use Task methods

### PR DESCRIPTION
**Pull request by stephentoub**
Remove Helpers.* and use Task.* methods instead.  Also use the public Task.FromCanceled instead of the internals-visible-to FromCancellation.

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @jkotas, @mellinoe

**Comment by JonHanna** [comment]
:+1:

File: src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs [hunk 1]
```diff
@@ -43,38 +43,6 @@ internal static void TryDeregister(this CancellationTokenRegistration ctr)
...
-        {
-            var taskSource = new TaskCompletionSource<TYPE>();
-            taskSource.SetCanceled();
-            taskSource.Task.Wait(cancellationToken);
-            return taskSource.Task;
-        }
-
         internal unsafe static void ZeroMemory(byte* src, long len)
         {
             while (len-- > 0)
```
Comment (benaadams) [inline src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs]: Don't know if its a hot path; but should this operate on a larger size than byte (e.g. long/int) when size is larger?

Comment (stephentoub) [inline src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs]: Potentially, but a) it's unrelated to this PR and I'm not planning to change it here, b) this helper is used only in an internal copy of UnmanagedMemoryStream in System.Runtime.WindowsRuntime and only in some relatively rare circumstances (e.g. when writing beyond the current end of the stream and needing to zero out memory between the current end and the write), and c) I'd hope that this private copy could be removed entirely in favor of just using System.IO.UnmanagedMemoryStream.dll in corefx... so updating this helper to be more clever doesn't seem particularly high priority.  You're welcome to open an issue to track it, though, if you feel differently.

Comment (benaadams) [inline src/System.Runtime.WindowsRuntime/src/System/InternalHelpers.cs]: No; was mostly curious. Thank you :)

File: src/System.Runtime.WindowsRuntime/src/System/WindowsRuntimeSystemExtensions.cs [hunk 2]
```diff
@@ -303,13 +303,13 @@ public static Task AsTask<TProgress>(this IAsyncActionWithProgress<TProgress> so
             switch (source.Status)
             {
                 case AsyncStatus.Completed:
-                    return Helpers.CompletedTask;
+                    return Task.CompletedTask;
 
                 case AsyncStatus.Error:
-                    return Helpers.TaskFromException<VoidValueTypeParameter>(source.ErrorCode.AttachRestrictedErrorInfo());
+                    return Task.FromException<VoidValueTypeParameter>(source.ErrorCode.AttachRestrictedErrorInfo());
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System/WindowsRuntimeSystemExtensions.cs]: Use the non-generic overload instead of `VoidValueTypeParameter`?

**Comment by jkotas** [comment]
LGTM otherwise

File: src/System.Runtime.WindowsRuntime/src/System/WindowsRuntimeSystemExtensions.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Runtime.WindowsRuntime/src/System/WindowsRuntimeSystemExtensions.cs]: Yup, fixed. Thanks.

**Pull request merged**